core/src/replay_stage.rs

```diff
@@ -3467,7 +3472,16 @@ impl ReplayStage {
                         total_stake,
                         switch_fork_decision,
                     );
-                    reset_bank.map(|b| (b, final_switch_fork_decision))
+                    let candidate_vote_bank = if final_switch_fork_decision.can_vote() {
+                        candidate_vote_bank
+                    } else {
+                        None
+                    };
+                    CandidateVoteAndResetBanks {
+                        candidate_vote_bank,
+                        reset_bank: candidate_vote_bank,
+                        switch_fork_decision: final_switch_fork_decision,
+                    }
                 }
```

Comment:
@wen-coding this is the trickiest case because it is the only time the `candidate_vote_bank` can potentially not be the `heaviest_bank` because of the vote refresh onto the `heaviest_bank_on_same_voted_fork`, so it would be good to get your eyes to verify this is correct